src/target/linux.rs

```diff
@@ -218,6 +277,103 @@ impl LinuxTarget {
 
         Ok(tasks)
     }
+
+    #[cfg(target_arch = "x86_64")]
+    pub fn set_watchpoint(
+        &mut self,
+        watchpoint: Watchpoint,
+    ) -> Result<usize, Box<dyn std::error::Error>> {
+        let empty = self.find_empty_watchpoint();
+        if empty.is_none() {
+            return Err(Box::new(WatchpointError::NoEmptyWatchpoint));
+        }
+        let index = empty.unwrap();
+
+        let rw_bits: u32 = match watchpoint.typ {
+            WatchpointType::Execute => 0b00,
+            WatchpointType::Read => 0b11,
+            WatchpointType::ReadWrite => 0b11,
+            WatchpointType::Write => 0b01,
+        } << 16 + index * 2;
+
+        let size_bits: u32 = (watchpoint.size as u32) << 18 + index * 2;
+
+        let enable_bit: u32 = 1 << (2 * index);
+
+        let bit_mask: u32 = (0b11 << (2 * index)) | (0b1111 << (16 + 4 * index));
+
+        let mut dr7: u64 = 0;
+
+        unsafe {
+            self.read().read(&mut dr7, *DEBUG_REG + 7 * 4).apply()?;
```

Comment:
This needs `PTRACE_PEEKUSER` I think, not `PTRACE_PEEK`/`read_process_vm` like `read` uses.

Reply:
🤦 I somehow missed that. Thanks!